sources/api/sundog/src/main.rs

```diff
@@ -566,10 +614,22 @@ async fn run() -> Result<()> {
     }
 
     info!("Retrieving settings values");
-    let settings = get_dynamic_settings(&args.socket_path, generators).await?;
+
+    let (weak_settings, strong_settings) =
+        get_dynamic_settings(&args.socket_path, generators).await?;
+
+    trace!(
+        "Weak settings from dynamic settings function: {:#?}",
+        weak_settings
+    );
+    trace!(
+        "Strong settings from dynamic settings function: {:#?}",
+        strong_settings
+    );
 
     info!("Sending settings values to the API");
-    set_settings(&args.socket_path, settings).await?;
+    set_settings(&args.socket_path, weak_settings, Strength::Weak).await?;
+    set_settings(&args.socket_path, strong_settings, Strength::Strong).await?;
```

Comment:
nit: Should we check that the settings aren't empty before submitting them?

Reply:
It is struct of type model::Settings, hence can not check if it is empty or not?